src/features/Search/index.js

```diff
@@ -0,0 +1,159 @@
+  const focusSearch = useRef(null);
+
+  useEffect(() => {
+    if (props.show) {
+      focusSearch.current.focus();
+    }
+  }, [props.show]);
+
+  const changeSearchType = (event) => {
+    setSearchType(event);
+    setSearchQuery("");
+  };
+
+  const loadStockSymbols = (query) => {
+    return fetchStockSymbols(query);
+  };
+
+  const loadCurrencySymbols = (query) => {
+    return fetchCurrencySymbols(query);
+  };
+
+  const loadCryptoSymbols = (query) => {
+    return fetchCryptoSymbols(query);
+  };
+
+  const sleep = (ms) => {
+    return new Promise((resolve) => setTimeout(resolve, ms));
+  };
+
+  // Todo this shouldn't be happning in useEffect but rather onChange
+  useEffect(() => {
+    let currentQuery = true;
+    const controller = new AbortController();
+
+    const loadSearchResults = async () => {
+      if (!searchQuery) {
+        setIsLoading(false);
+        return setSearchResults([]);
+      }
+
+      setIsLoading(true);
+      setSearchResults([]);
+
+      await sleep(500);
+      switch (searchType) {
+        case searchTypes.STOCKS:
+          if (currentQuery) {
+            const stocks = loadStockSymbols(searchQuery, controller);
+            setSearchResults(stocks);
+            setIsLoading(false);
+          }
+          break;
+        case searchTypes.CURRENCIES:
+          if (currentQuery) {
+            const currencies = loadCurrencySymbols(searchQuery, controller);
+            setSearchResults(currencies);
+            setIsLoading(false);
+          }
+          break;
+        case searchTypes.CRYPTOCURRENCIES:
+          if (currentQuery) {
+            const cryptocurrencies = loadCryptoSymbols(searchQuery, controller);
+            setSearchResults(cryptocurrencies);
+            setIsLoading(false);
+          }
+          break;
+        default:
+          setSearchResults([]);
+          setIsLoading(false);
+      }
+    };
+
+    loadSearchResults();
+
+    return () => {
+      currentQuery = false;
+      controller.abort();
+    };
+  }, [searchQuery, searchType]);
+
+  return (
+    <Fragment>
+      <div className={styles.modalHeader}>
+        <NavigationItem
+          active={searchType === searchTypes.STOCKS}
+          text="Stocks"
+          onClick={() => changeSearchType(searchTypes.STOCKS)}
+        />
+        <NavigationItem
+            active={searchType === searchTypes.CURRENCIES}
+            text="Currencies"
+            onClick={() => changeSearchType(searchTypes.CURRENCIES)}
+        />
+        <NavigationItem
+            active={searchType === searchTypes.CRYPTOCURRENCIES}
+            text="Cryptocurrencies"
+            onClick={() => changeSearchType(searchTypes.CRYPTOCURRENCIES)}
+        />
+      </div>
+      <div className={styles.modalSearchField}>
+        <input
+            className={styles.modalSearchField_input}
+            type="text"
+            placeholder={`Enter your search here, e.g. ${utils.getPlaceholderText(searchType)}`}
```

Comment:
previously the placeholder was a part of the state of the feature, but it was actually dependent on the `searchType` which is already a part of the state so no need to have them boh as part of the state when one is depend on the other